src/main/java/edu/ie3/datamodel/io/source/csv/CsvDataSource.java

```diff
@@ -78,15 +78,33 @@ public CsvDataSource(String csvSep, String folderPath, FileNamingStrategy fileNa
     TreeMap<String, String> insensitiveFieldsToAttributes =
         new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
 
-    final String[] fieldVals = fieldVals(csvSep, csvRow);
+    // when replacing deprecated workaround add final modifier before parseCsvRow as well as remove
+    // 'finalFieldVals' below!
+    String[] fieldVals = parseCsvRow(csvSep, csvRow);
+
+    // start workaround for deprecated data model processing
+    if (fieldVals.length != headline.length) {
+      // try to parse old structure
+      fieldVals = oldFieldVals(csvSep, csvRow);
+      // if this works log a warning to inform the user that this will not work much longer,
+      // otherwise
+      // parsing will fail regularly as expected below
+      if (fieldVals.length == headline.length)
+        log.warn(
+            "You are using an outdated version of the data "
+                + "model with invalid formatted csv rows. This is okay for now, but please updated your files, as the "
+                + "support for the old model will be removed soon.");
```

Comment:
What is if the `fieldVals` still don't have the same length? Is this covered well? Are lines 102...107 failing in any case? I think the case, where the entries are shorter as the head line is not covered well, isn't it?

Reply:
Understood now, what happens.